File: 2. Задачи с ЕГЭ по информатике/G. Пара с делимостью.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
	int n;
	int del_7 = 0; 
	int max = 0;
	while (true) {

		cin >> n;

		if (n == 0) break;

		if ((n % 7 == 0 && n % 49) && (n > del_7)) del_7 = n;
		else if ((n % 7 != 0) && (n > max)) max = n;
	}
	if (del_7 && max) cout << max * del_7;
	else cout << 1;
	return 0;
}

File: 2. Задачи с ЕГЭ по информатике/G. Пара с делимостью.py
DEL_7, MAX = 0, 0
while True:
    n = int(input())
    if n == 0:
        break
    if not n % 7 and n % 49 and n > DEL_7:
        DEL_7 = n
    elif n % 7 and n > MAX:
        MAX = n
if DEL_7 and MAX:
    print(DEL_7 * MAX)
else:
    print('1')